### ISSUE TITLE
"knife site share" not using chefignore when using cookbook-path toggle.

### ISSUE THREAD

**Issue by joekiller** [chef/chef#6379]
## Description

If I have a cookbook in an arbitrary directory and use the following to share, the chefignore in the cookbook isn't honored.

    knife cookbook site share <cookbook> --cookbook-path $(pwd)/..

## Chef Version

Chef: 13.2.20

## Platform Version

OSX 16.7.0

## Replication Case

  1. Clone https://github.com/adamsb6/s3_file
  1. Run `knife cookbook site share s3_file --cookbook-path $(pwd)/.. --dry-run`
  1. Output shows test files are being included even with the pattern `test/*` being in chefignore
  1. Copy chefignore to $(pwd)/.. and it is used as expected.

## Client Output

```
$ knife cookbook site share s3_file --cookbook-path $(pwd)/.. --dry-run
Generating metadata for s3_file from /var/folders/k1/dy0j4z_16lj16p8940w452vm0000gn/T/chef-s3_file-build20170906-1157-6hh0d5/s3_file/metadata.rb
Making tarball s3_file.tgz
Not uploading s3_file.tgz due to --dry-run flag.
s3_file/
s3_file/.gitignore
s3_file/.kitchen.yml
s3_file/attributes/
s3_file/Berksfile
s3_file/bin/
s3_file/CHANGELOG.md
s3_file/chefignore
...
s3_file/test/fixtures/
...
s3_file/test/fixtures/cookbooks/s3_file_test/recipes/default.rb
s3_file/resources/default.rb
...
s3_file/attributes/default.rb
```

**Comment by DecoyJoe** [comment]
This issue is a duplicate of issue https://github.com/chef/chef/issues/2994, is being fixed in PR https://github.com/chef/chef/pull/7156

**Comment by vsingh-msys** [comment]
Fixed in #8925
